### DISCUSSION TITLE
Sales channel won't be attached to the API key

### DISCUSSION THREAD

**Discussion by jake1808** [medusajs/medusa#4823] [Q&A]
Hello again!! I made a sales channel and followed the steps to create a API key. after creating it I noticed that the sales channel I assigned to that API key wasn't registered I wen in the DB and inserted a row in the `publishable_api_key_sales_channel` table with the appropriate IDs. I checked the admin panel and my channel was attached to the API key. How ever when I try to use the API key in my front-end Nextjs template it still brings back the default sales channel.

**Comment by fPolic** [comment]
Hey @jake1808, I couldn't reproduce the first issue you mentioned, could you share more details?

Which endpoint on the storefront is bringing the default channel? 
The idea behind PublishableAPI keys is to scope products and carts. Sales channels aren't scoped by API keys. 
Sales channels are managed on the admin and their endpoints shouldn't be called from the store.

The use case for Publishable API keys would be:
1. Create a sales channel
2. Add products to the channel
3. Create an API key and associate it with the channel

Now, if you register the key in your storefront and call for example `/store/products` you would only get products from the associated sales channel.
